network/simple-onchain-discovery/src/lib.rs

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use futures::sink::SinkExt;
+use futures::{select, StreamExt};
+use once_cell::sync::Lazy;
```

Comment:
nit on ordering, while there is something more reasonable about your approach, it would result in much bike shedding :P

Reply:
no to bike shedding.  divergence from standard is not intended here.